packages/webpack-cli/lib/plugins/WebpackCLIPlugin.js

```diff
@@ -1,63 +1,120 @@
 const packageExists = require('../utils/package-exists');
 const webpack = packageExists('webpack') ? require('webpack') : undefined;
-const logger = require('../utils/logger');
-
-const PluginName = 'webpack-cli';
+const { getStatsOptions } = require('../utils/stats-options');
+const { PluginName } = require('../utils/name');
 
 class WebpackCLIPlugin {
     constructor(options) {
         this.options = options;
+
+        this.isWatch = false;
     }
 
     async apply(compiler) {
-        const compilers = compiler.compilers || [compiler];
+        const { progress } = this.options;
+        if (progress) {
+            this.appendProgressPlugin(compiler, progress);
+        }
 
-        for (const compiler of compilers) {
-            if (this.options.progress) {
-                const { ProgressPlugin } = compiler.webpack || webpack;
+        const isWebpack5 = Boolean(compiler.webpack);
 
-                let progressPluginExists;
+        const logger = compiler.getInfrastructureLogger(PluginName);
 
-                if (compiler.options.plugins) {
-                    progressPluginExists = Boolean(compiler.options.plugins.find((plugin) => plugin instanceof ProgressPlugin));
-                }
+        const resolveName = (obj) => (obj.name ? `${obj.name} ` : '');
 
-                if (!progressPluginExists) {
-                    if (typeof this.options.progress === 'string' && this.options.progress !== 'profile') {
-                        logger.error(
-                            `'${this.options.progress}' is an invalid value for the --progress option. Only 'profile' is allowed.`,
-                        );
-                        process.exit(2);
-                    }
+        const done = (stats) => {
+            const printStats = (childCompiler, childStats) => {
+                const name = resolveName(childCompiler);
+
+                const status = childStats.toString({
+                    all: false,
+                    version: true,
+                    timings: true,
+                });
+
+                const normalizedStatus = (() => {
+                    if (isWebpack5) {
+                        return status;
+                    } else {
+                        const [version, time] = status.split('\n').map((line) => {
+                            const value = line.split(':')[1] || '';
+                            return value.trim();
+                        });
 
-                    const isProfile = this.options.progress === 'profile';
+                        return `${name ? name + `(${version})` : version} compiled in ${time}`;
+                    }
+                })();
 
-                    new ProgressPlugin({ profile: isProfile }).apply(compiler);
+                if (childStats.hasErrors()) {
+                    logger.error(normalizedStatus);
```

Comment:
We should set exitCode = 1 for errors.

Reply:
Exit code is set in callback in webpack-cli.js this wasn't changed https://github.com/webpack/webpack-cli/pull/2036/files#diff-7c711e2a4ef4b16d2cbbe1658495723402f9692da654cb6fa959127f3dc779f2R357-R359

Wondering if we should reset it if next complication is has no errors 🤔